relay-sampling/src/lib.rs

```diff
@@ -84,1522 +84,39 @@ pub use dsc::DynamicSamplingContext;
-            replay_id: Some(Uuid::new_v4()),
-            environment: Some("debug".to_string()),
-            transaction: Some("transaction1".into()),
-            sample_rate: None,
-            sampled: None,
-            other: BTreeMap::new(),
-        };
-
-        assert!(
-            condition.matches(&dsc),
-            "did not match with missing user segment"
-        );
-
-        let condition = and(vec![
-            glob("trace.release", &["1.1.1"]),
-            eq("trace.user.segment", &["vip"], true),
-        ]);
-        let dsc = DynamicSamplingContext {
-            trace_id: Uuid::new_v4(),
-            public_key: ProjectKey::parse("abd0f232775f45feab79864e580d160b").unwrap(),
-            release: Some("1.1.1".to_string()),
-            user: TraceUserContext {
-                user_segment: "vip".to_owned(),
-                user_id: "user-id".to_owned(),
-            },
-            replay_id: None,
-            sampled: None,
-            environment: None,
-            transaction: Some("transaction1".into()),
-            sample_rate: None,
-            other: BTreeMap::new(),
-        };
-
-        assert!(
-            condition.matches(&dsc),
-            "did not match with missing environment"
-        );
-
-        let condition = and(vec![
-            glob("trace.release", &["1.1.1"]),
-            eq("trace.user.segment", &["vip"], true),
-        ]);
-        let dsc = DynamicSamplingContext {
-            trace_id: Uuid::new_v4(),
-            public_key: ProjectKey::parse("abd0f232775f45feab79864e580d160b").unwrap(),
-            release: Some("1.1.1".to_string()),
-            user: TraceUserContext {
-                user_segment: "vip".to_owned(),
-                user_id: "user-id".to_owned(),
-            },
-            replay_id: None,
-            sampled: None,
-            environment: Some("debug".to_string()),
-            transaction: None,
-            sample_rate: None,
-            other: BTreeMap::new(),
-        };
-
-        assert!(
-            condition.matches(&dsc),
-            "did not match with missing transaction"
-        );
-        let condition = and(vec![]);
-        let dsc = DynamicSamplingContext {
-            trace_id: Uuid::new_v4(),
-            public_key: ProjectKey::parse("abd0f232775f45feab79864e580d160b").unwrap(),
-            release: None,
-            user: TraceUserContext::default(),
-            replay_id: None,
-            sampled: None,
-            environment: None,
-            transaction: None,
-            sample_rate: None,
-            other: BTreeMap::new(),
-        };
-
-        assert!(
-            condition.matches(&dsc),
-            "did not match with missing release, user segment, environment and transaction"
-        );
-    }
-
-    #[test]
-    /// test that the multi-matching returns none in case there is no match.
-    fn test_multi_matching_with_transaction_event_non_decaying_rules_and_no_match() {
```

Comment:
removed because its already tested in     `fn test_get_sampling_match_result_with_no_match()`